### PR TITLE
Fix switch statement when handling IQ stanza for registration

### PR DESCRIPTION
**Pull request by melvo**

### CODE REVIEW

**Comment by lnjX** [comment]
Thank you.

Now we also need to improve the unit tests to prevent that something like this happens again. Could you do that?

**Comment by lnjX** [comment]
Could you change the base branch to `release/1.2.1`, so we can make a patch release with this?

**Comment by melvo** [comment]
The base branch is changed and the test added.

File: tests/qxmppregistrationmanager/tst_qxmppregistrationmanager.cpp [hunk 1]
```diff
@@ -370,7 +370,14 @@ void tst_QXmppRegistrationManager::testRegistrationResult()
...
     manager->handleStanza(writePacketToDom(serverResult));
 
-    QVERIFY(signalCalled);
+    if (isSuccess) {
+        QVERIFY(succeededSignalCalled);
+        QVERIFY(!failedSignalCalled);
+    } else {
+        QVERIFY(!succeededSignalCalled);
+        QVERIFY(failedSignalCalled);
+    }
```
Comment (lnjX) [inline tests/qxmppregistrationmanager/tst_qxmppregistrationmanager.cpp]: I think this is even better:
```c++
    QCOMPARE(succeededSignalCalled, isSuccess);
    QCOMPARE(failedSignalCalled, !isSuccess);
```

**Pull request merged**